### PR TITLE
Update GPT2 Model Benchmark Script to Support IO Binding

### PR DESCRIPTION
**Pull request by liuziyue**
**Description**: Describe your changes.
Add an option to run onnxruntime with io binding for benchmark. 

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.
With IO binding, inputs and outputs are preallocated in memory, and copying across devices is avoided. Therefore for gpt2 model on GPU, the latency of ort is greatly improved .

### CODE REVIEW

File: onnxruntime/python/tools/transformers/benchmark_gpt2.py [hunk 1]
```diff
@@ -301,14 +376,30 @@ def main():
...
+            last_state_shape = [
+                batch_size, 1, config.hidden_size
+            ]
+            present_shape = [
+                2, batch_size, config.num_attention_heads, sequence_length + 1, 
+                int(config.hidden_size / config.num_attention_heads)
+            ]
+
+            latencies = inference(model,
+                                    session,
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/benchmark_gpt2.py:390]: format this script (see onnxruntime\ReformatSourcePython.bat)

File: onnxruntime/python/tools/transformers/benchmark_gpt2.py [hunk 2]
```diff
@@ -76,26 +76,73 @@ def onnxruntime_inference(ort_session, input_ids, past=None, total_runs=100):
 
     return ort_outputs, average_latency
 
+def onnxruntime_inference_with_binded_io(ort_session, input_ids, last_state, last_state_shape, past=None, present=None, present_shape=None, total_runs=100):
+    # Bind inputs and outputs to onnxruntime session        
+    io_binding = ort_session.io_binding()
+    # Bind inputs
+    io_binding.bind_input('input_ids', input_ids.device.type, 0, numpy.longlong,
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/benchmark_gpt2.py:83]: It is better to use numpy.int64 to be consistent with onnx data type.

Comment (liuziyue) [reply onnxruntime/python/tools/transformers/benchmark_gpt2.py]: For some reason numpy.int64 will result in "RuntimeError: No corresponding Numpy type for Tensor Type." error here. It seems like int64 type is matched to numpy.longlong in onnxruntime/python/onnxruntime_pybind_mlvalue.cc

File: onnxruntime/python/tools/transformers/benchmark_gpt2.py (hunk 1 above)
Comment (liuziyue) [reply onnxruntime/python/tools/transformers/benchmark_gpt2.py]: Formatted.

File: onnxruntime/python/tools/transformers/benchmark_gpt2.py [hunk 3]
```diff
@@ -336,12 +414,16 @@ def main():
         onnx_model_path = os.path.join(output_dir, 'gpt2_past_optimized.onnx')
         m.save_model_to_file(onnx_model_path)
 
-    if args.use_gpu and 'CUDAExecutionProvider' not in onnxruntime.get_available_providers():
-        logger.warning("Please install onnxruntime-gpu package to test GPU inference.")
+    if args.use_gpu and 'CUDAExecutionProvider' not in onnxruntime.get_available_providers(
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/benchmark_gpt2.py:417]: Did you run the reformatpython.bat? It seems that line length limit is less than 120 in your format.

File: onnxruntime/python/tools/transformers/benchmark_gpt2.py (hunk 2 above)
Comment (tianleiwu) [reply onnxruntime/python/tools/transformers/benchmark_gpt2.py]: I see. Then let's keep longlong.

File: onnxruntime/python/tools/transformers/benchmark_gpt2.py (hunk 1 above)
Comment (tianleiwu) [reply onnxruntime/python/tools/transformers/benchmark_gpt2.py]: Did you run the reformatpython.bat? It seems that line length limit is less than 120 in your format.

Comment (liuziyue) [reply onnxruntime/python/tools/transformers/benchmark_gpt2.py]: Fixed format. The default values in my visual studio code was different.  

Also I found out that the https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/.style.yapf "base_on_style" has been changed from google to pep8. So I modified the instructions in the reformatpython.bat for consistency.

**Pull request merged**